### PR TITLE
Upgrade MicroPython to 2.2.1 and 1.1.1.

### LINKED ISSUE #1071: Upgrade MicroPython
New releases:
- https://github.com/bbcmicrobit/micropython/releases/tag/v1.1.1
- https://github.com/microbit-foundation/micropython-microbit-v2/releases/tag/v2.1.1

Let's use this as a chance to document the process.

Simulator:

- [x] Review changelogs and diff to determine simulator changes, stubs changes any other documentation changes
- [x] Raise educational content impact to Foundation education team (likely already in hand or low priority)
- [x] Update simulator as required
- [x] Run through simulator examples in simulator demo.html
- [x] Test simulator in Python Editor (locally)
- [x] Merge simulator changes

Editor:

- [x] Update stubs as required
- [x] Update MicroPython hex files and version metadata in the editor
- [x] PR with updated sim (via staging) and stubs
- [x] Release stubs
- [ ] Test and merge PR

Now or later:

- [ ] Release simulator
- [ ] Update and publish simulator versioning CloudFront function
- [ ] Release editor

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Stubs changes are on https://github.com/microbit-foundation/micropython-microbit-stubs/pull/85

Closes #1071
Closes https://github.com/microbit-foundation/python-editor-v3/issues/1001

### CODE REVIEW

**Comment by microbit-matt-hillsdon** [comment]
Confirmed https://github.com/microbit-foundation/python-editor-v3/issues/1001 is fixed on the review branch.

File: .github/ISSUE_TEMPLATE/micropython-update.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+name: Upgrade MicroPython
+about: Template for MicroPyton upgrades
```
Comment (microbit-matt-hillsdon) [inline .github/ISSUE_TEMPLATE/micropython-update.md:3]:
```suggestion
about: Template for MicroPython upgrades
```

File: src/simulator/Simulator.tsx [hunk 2]
```diff
@@ -43,11 +43,11 @@ const Simulator = ({
...
 }: SimulatorProps) => {
-  const version = "0.1";
-  const url = `https://python-simulator.usermbit.org/v/${version}/simulator.html`;
+  //const version = "0.1";
+  //const url = `https://python-simulator.usermbit.org/v/${version}/simulator.html`;
   // For testing with sim branches:
-  //const branch = "whatever";
-  //const url = `https://review-python-simulator.usermbit.org/${branch}/simulator.html`;
+  const branch = "upgrade-mpy";
+  const url = `https://review-python-simulator.usermbit.org/${branch}/simulator.html`;
```
Comment (microbit-matt-hillsdon) [inline src/simulator/Simulator.tsx:50]:
```suggestion
  const version = "0.1";
  const url = `https://python-simulator.usermbit.org/v/${version}/simulator.html`;
  //const branch = "upgrade-mpy";
  //const url = `https://review-python-simulator.usermbit.org/${branch}/simulator.html`;
```

**Review by microbit-robert — APPROVED** [review]
LGTM.

**Comment by microbit-matt-hillsdon** [comment]
Tested V1 and V2 flashing. Rob and I checked the unicode flag fix and all looks good in the sim.

**Pull request merged**